src/Stores/Cases/actions.js

```diff
@@ -0,0 +1,64 @@
+/* @flow */
+import type { Dispatch } from '../../types';
+import type { Case, CaseId } from './types';
+import actionTypes from './actionTypes';
+
+import {
+  getParliamentCaseById as apiGetParliamentCaseById,
+  getParliamentCases as apiGetParliamentCases,
+} from '../../utils/api';
+
+/* 
+    GET CASE BY ID
+*/
+
+const getCaseById = () => ({ type: actionTypes.GET_CASE_BY_ID_REQUEST });
+
+const getCaseByIdSuccess = (parliamentCase: Case) => ({
+  case: parliamentCase,
+  type: actionTypes.GET_CASE_BY_ID_SUCCESS,
+});
+
+const getCaseByIdFailure = (error: *) => ({
+  error,
+  type: actionTypes.GET_CASE_BY_ID_FAILURE,
+});
+
+export const fetchCaseById = (id: CaseId) => (dispatch: Dispatch) => {
+  dispatch(getCaseById());
+
+  return apiGetParliamentCaseById(id).then(
+    resp => resp.data && dispatch(getCaseByIdSuccess(resp.data)),
+    error => dispatch(getCaseByIdFailure(error.message))
```

Comment:
I messed up the API <-> actions interface here.

Because the API does 
```
 .then(checkStatus)
    .then(parseJSON)
    .then(mapData)
    .catch(error => ({ error })) <---- this will make it so the API does not throw
```

We can fix it in an upcoming PR or fix it here.

the fix is to either
1. let the API throw errors and catch them in the Actions 
2. Actions should check if there is an error

Solution 1. means changing the API
Solution 2. means changing the actions to something like this:
```
  return apiGetParliamentCaseById(id).then(resp => {
    if (resp.error) {
      error => dispatch(getCaseByIdFailure(error.message));
      return;
    }
    resp.data && dispatch(getCaseByIdSuccess(resp.data))
```